keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -161,146 +159,157 @@ consistent with the syntax supported by any of the common proxy vendors:
+Interoperability between versions will be as follows:
 
-2) v1beta1 gets a "PathPrefix" field as above.  v1 gets an annotation
-"ingress.kubernetes.io/beta-path-semantics".  If we create a v1 object
-from a v1beta1 where `path` was used, we set the annotation.
-Validation logic will check the annotation to decide how to interpret
-and validate paths.  We'll have to keep it for a long while.  Maybe we
-can prevent new objects from being created with that annotation?  Now
-what if implementations want to offer "extended" semantics here?  Same
-validation problem.
+1. v1beta1 will default to `ImplementationSpecific`.
+1. v1 will default to `Prefix`.
 
-3)  v1beta1 gets a "PathPrefix" field as above.  v1 gets an annotation
-"ingress.kubernetes.io/non-portable-path-semantics".  If we create a
-v1 object from a v1beta1 where `path` was used, we set the annotation.
-We keep it forever.
+Note: default value are permitted between API versions
+[ref][api-conv-versions].
 
-4) Same as 2 but we just don't validate Path at all.
--->
+[api-conv-versions]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#defaulting
 
-#### Rejected alternatives
+##### Prefix match semantics
 
-##### Portable regex behavior
+The semantics of a prefix match is as follows:
 
-The safest route for specifying the regex would be to state a limited
-subset that can be used in a portable way. Any expressions outside of
-the subset will have implementation specific behavior.
+* Let `(p_1, p_2, ..., p_n)` be the list of paths for a specific host.
+* Every `p_i` must begin with the `'/'` character (relative paths are
+  not allowed by [RFC-7230][rfc7230]0.
+* A request is a potential match for path `p_i` if `p_i` is the prefix of
+  the path. If there is more than one potential match, the longest
+  `p_i` will be the matching path.
+* If there is no matching path, then the `defaultBackend` for the host
+  will be used. If there is not match for the host, then the overall
+  `defaultBackend` for the Ingress will be selected.
 
-Regular expression subset (derived from [re2][re2-syntax] syntax page)
-
-| Expression | description             |
-|------------|-------------------------|
-| `.`        | any character           |
-| `[xyz]`    | character class         |
-| `[^xyz]`   | negated character class |
-| `x*`       | 0 or more x's           |
-| `x+`       | 1 or more x's           |
-| `xy`       | x followed by y         |
-| `x|y`      | x or y (prefer x)       |
-| `(abc)`    | grouping                |
-
-Maintaining a regular expression subset is likely not worth the complexity and
-is likely impossible across the [many implementations][regex-survey].
+[rfc7230]: https://tools.ietf.org/html/rfc7230#section-5.3.1
 
 ### API field naming
 
-These are straightforwarding one-to-one renames to for consistency.
+These are straightforwarding one-to-one renames for better semantic
+meaning.
```

Comment:
This may be difficult to resolve, as the notion of an overall default backends maybe incompatible with sharing of a single VIP among different Ingress objects. I would leave the behavior up to the implementation in this case if there is such a conflict.

Reply:
Added a clarifying comment.